packages/applications-service-api/src/controllers/advice.js

```diff
@@ -2,78 +2,83 @@ const { StatusCodes } = require('http-status-codes');
 
 const logger = require('../lib/logger');
 const { documentsHost } = require('../lib/config');
-const { getAdvice, getAdviceById } = require('../services/advice.service');
-
+const {
+	getAllAdvice: getAllAdviceService,
+	getAdviceById: getAdviceByIdService
+} = require('../services/advice.service');
 const ApiError = require('../error/apiError');
 
-module.exports = {
-	async getAdvice(req, res) {
-		logger.debug(`Retrieving advice ...`);
-		try {
-			const itemsPerPage = Math.min(Number(req.query.size), 100) || 25;
-			const page = Number(req.query.page) || 1;
+const getAdvice = async (req, res) => {
+	logger.debug(`Retrieving advice ...`);
+	const { caseRef } = req.query;
 
-			const advice = await getAdvice({
-				caseReference: req.query.caseRef,
-				page,
-				itemsPerPage,
-				searchTerm: req.query.searchTerm
-			});
+	try {
+		if (!caseRef) {
+			throw ApiError.badRequest('missing required parameter: caseRef');
+		}
 
-			const paginationData = {
-				totalItems: advice.count,
-				itemsPerPage: itemsPerPage,
-				totalPages: Math.ceil(Math.max(1, advice.count) / itemsPerPage),
-				currentPage: page
-			};
+		const { advice, totalItems, itemsPerPage, totalPages, currentPage } = await getAllAdviceService(
+			req.query
+		);
 
-			res.status(StatusCodes.OK).send({
-				advice: advice.rows,
-				...paginationData
-			});
-		} catch (e) {
-			if (e instanceof ApiError) {
-				logger.debug(e.message);
-				res.status(e.code).send({ code: e.code, errors: e.message.errors });
-				return;
-			}
-			logger.error(e.message);
-			res.status(StatusCodes.INTERNAL_SERVER_ERROR).send({
-				message: `Problem getting advice \n ${e}`
-			});
+		if (totalItems === 0) {
```

Comment:
Do we want this behaviour of throwing an error if the caseref is valid but there's no advice? This `adviceNotFound` error message seems more for the scenario where you look up Advice by ID and a record with the given ID isn't found. 

Unless we have an AC for changing the behaviour to return 404, then returning 200 with an empty array is more in line with other endpoints

Reply:
Good point, I wasn't too sure, I did this to follow what the timetable was doing (which was updated by me recently but had this beforehand too). I don't know where we would throw `adviceNotFound`, would we be searching to see if the case ref exists in the project table to confirm that the case exists but we found no advice (this sounds may be over complicated)?

Reply:
I think it makes sense for the `adviceNotFound` error to remain in the `getAdviceById` function, so it returns 404 if the given `id` for `GET /advice/{id}` is not found in Advice table.

For the `GET /advice` endpoint, I think it makes sense to return 400 when caseRef is missing (as you already are), but I wouldn't go as far as validating that caseRef exists in the Projects table. So if we remove the `if (totalItems === 0) { throw ApiError.adviceNotFound(caseRef)` condition, I think that covers all the ACs without regression or introducing new behaviour